File: 5-Problem-A-Day/134. Gas Station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;

        int total = 0, curr = 0, start = 0;
        for (int i = 0; i < n; i++) {
            total += gas[i] - cost[i];
            curr += gas[i] - cost[i];
            if (curr < 0) {
                start = i + 1;
                curr = 0;
            }
        }
        return total >= 0 ? start : -1;
    }
}

File: 2024/134. Gas Station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int totalTank = 0, currTank = 0, cand = 0;
        for (int i = 0; i < gas.size(); i++) {
            totalTank += gas[i] - cost[i];
            currTank += gas[i] - cost[i];
            if (currTank < 0) {
                cand = (i + 1) % gas.size();
                currTank = 0;
            }
        }
        return totalTank >= 0 ? cand : - 1;
    }
};